core/src/main/scala/org/apache/spark/util/SparkUncaughtExceptionHandler.scala

```diff
@@ -36,6 +38,17 @@ private[spark] class SparkUncaughtExceptionHandler(val exitOnUncaughtException:
     val _ = SparkExitCode.OOM
   }
 
+  // The maximum depth to search in the exception cause chain for a fatal error,
+  // as defined by killOnFatalErrorDepth in Executor.scala.
+  //
+  // SPARK-50034: When this handler is called, there is a fatal error in the cause chain within
+  // the specified depth. We should identify that fatal error and exit with the
+  // correct exit code.
+  private val killOnFatalErrorDepth: Int =
+    // At this point SparkEnv might be None
+    Option(SparkEnv.get).map(_.conf.get(EXECUTOR_KILL_ON_FATAL_ERROR_DEPTH)).getOrElse(5)
```

Comment:
Why do we have EXECUTOR in EXECUTOR_KILL_ON_FATAL_ERROR_DEPTH name? We are sometimes killing master\worker depending upon the value of EXECUTOR_KILL_ON_FATAL_ERROR_DEPTH as master and worker use SparkUncaughtExceptionHandler.

Reply:
Yes, let's remove the executor from the name? SparkUncaughtExceptionHandler is also used by master.

Reply:
Sounds good, I'll remove "EXECUTOR" from the name